tests/test_thread.cpp

```diff
@@ -0,0 +1,64 @@
+/*
+    tests/test_thread.cpp -- call pybind11 bound methods in threads
+
+    Copyright (c) 2017 Laramie Leavitt (Google LLC) <lar@google.com>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#include <chrono>
+#include <thread>
+
+#include <pybind11/cast.h>
+#include <pybind11/pybind11.h>
+
+#include "pybind11_tests.h"
+
+
+namespace py = pybind11;
+
+
+namespace {
+
+struct IntStruct {
+    int value;
+};
+
+}
+
+TEST_SUBMODULE(thread, m) {
+
+    py::class_<IntStruct>(m, "IntStruct")
+        .def(py::init([](const int i) { return IntStruct{i}; }));
+
+    // implicitly_convertible uses loader_life_support when an implicit
+    // conversion is required in order to llifetime extend the reference.
+    py::implicitly_convertible<int, IntStruct>();
+
+
+    m.def("test", [](const IntStruct& in) {
+        IntStruct copy = in;
+
+        {
+          py::gil_scoped_release release;
+          std::this_thread::sleep_for(std::chrono::milliseconds(5));
+        }
+
+        if (in.value != copy.value) {
+            throw std::runtime_error("Reference changed!!");
+        }
+    });
+
+   m.def("test_no_gil", [](const IntStruct& in) {
+        IntStruct copy = in;
+        std::this_thread::sleep_for(std::chrono::milliseconds(5));
+        if (in.value != copy.value) {
+            throw std::runtime_error("Reference changed!!");
+        }
+    },
+    py::call_guard<py::gil_scoped_release>());
+
+    // NOTE: std::string_view also uses loader_life_support to ensure that
```

Comment:
You can test string_view too if you want using the appropiate if_defs (see our current stringview tests)

Reply:
It shouldn't be necessary here.